docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
-# Developer Guide
+# Developer Guide for Yet Another Module Organiser / Manager
 
-## Acknowledgements
+## Table Of Contents
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [Developer Guide for Yet Another Module Organiser / Manager](#developer-guide-for-yet-another-module-organiser--manager)
+  - [Table Of Contents](#table-of-contents)
+  - [1. Introduction](#1-introduction)
+    - [1.1. Welcome!](#11-welcome)
+  - [2. Setting up](#2-setting-up)
+    - [2.1. Prerequisites](#21-prerequisites)
+    - [2.2. Setting up the project locally](#22-setting-up-the-project-locally)
+    - [2.3. Configure the coding style](#23-configure-the-coding-style)
+  - [3. Design](#3-design)
+    - [3.1 Architecture](#31-architecture)
+    - [3.2 Model Component](#32-model-component)
+    - [3.3 Parser Component](#33-parser-component)
+    - [3.4 Command Component](#34-command-component)
+    - [3.5 Utils Component](#35-utils-component)
+      - [3.5.1 UI Component](#351-ui-component)
+      - [3.5.2 Link Component](#352-link-component)
+      - [3.5.3 Storage Component](#353-storage-component)
+        - [Why it is implemented this way](#why-it-is-implemented-this-way)
+        - [Alternatives considered](#alternatives-considered)
+  - [4. Implementation](#4-implementation)
+    - [Storage feature](#storage-feature)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+  - [5. Documentation](#5-documentation)
+  - [6. Testing](#6-testing)
+  - [6.1. Running tests](#61-running-tests)
+  - [6.2 Instructions for manual testing](#62-instructions-for-manual-testing)
+  - [Appendix A: Product scope](#appendix-a-product-scope)
+  - [Appendix B: User Stories](#appendix-b-user-stories)
+  - [Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)
+  - [Appendix D: Glossary](#appendix-d-glossary)
+  - [Appendix E: Acknowledgements](#appendix-e-acknowledgements)
+    - [Third-party libraries](#third-party-libraries)
 
```

Comment:
Seems like the appendices below are duplicated. Both lead to the same component, maybe can delete the bottom one?